app/services/supplementary-billing/abstraction-billing-period.service.js

```diff
@@ -137,23 +139,61 @@ function _abstractionPeriods (billingPeriod, chargePurpose) {
 }
 
 /**
- * Calculates the 'billable days' for each abstraction billing period
+ * Calculates the 'billable days' for each abstraction billable period
  *
  * @param {Object[]} abstractionPeriods An array of abstraction billing periods
  *
  * @returns {Object[]} The array abstraction periods each with a new `billableDays` property
  * @returns {Date} periods[].startDate
  * @returns {Date} periods[].endDate
  * @returns {boolean} periods[].consider
+ * @returns {Date} periods[].billableStartDate
+ * @returns {Date} periods[].billableEndDate
  * @returns {number} periods[].billableDays
  */
 function _calculateBillableDays (abstractionPeriods) {
-  let difference
-  let billableDays
   for (const abstractionPeriod of abstractionPeriods) {
-    difference = abstractionPeriod.endDate.getTime() - abstractionPeriod.startDate.getTime() // difference in msecs
-    billableDays = Math.ceil(difference / (1000 * 3600 * 24)) + 1 // (1000 msecs * (60 secs * 60 mins) * 24 hrs)
-    abstractionPeriod.billableDays = billableDays
+    if (abstractionPeriod.billableStartDate) {
+      const difference = abstractionPeriod.billableEndDate.getTime() - abstractionPeriod.billableStartDate.getTime() // difference in msecs
+      const billableDays = Math.ceil(difference / (1000 * 3600 * 24)) + 1 // (1000 msecs * (60 secs * 60 mins) * 24 hrs)
+      abstractionPeriod.billableDays = billableDays
+    }
+  }
+}
+
+/**
+ * Calculates the 'billable period' for each abstraction billing period
+ *
+ * @param {Object[]} abstractionPeriods An array of abstraction billing periods
+ * @param {Object} billingPeriod Object that has a `startDate` and `endDate` that defines the billing period
+ *
+ * @returns {Object[]} The array abstraction periods each with new `billableStartDate` & `billableEndDate` properties
```

Comment:
This applies to `_flagPeriodsForConsideration` as well, but we're saying that this function returns something when really it's modifying the object that's passed to it. More of an observation really as I don't know what the right way to document this behaviour would be!

Reply:
That is a valid point. That's my bad for getting trigger-happy with the JSDoc for `_flagPeriodsForConsideration ()`. Where we are not explicitly returning anything we shouldn't be adding `@returns` 🤦